core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -39,13 +39,21 @@
 
 @RunWith(Parameterized.class)
 public class TestRemoveSnapshots extends TableTestBase {
-  @Parameterized.Parameters(name = "formatVersion = {0}")
+  private final boolean incrementalCleanup;
+
+  @Parameterized.Parameters(name = "formatVersion = {0}, incrementalCleanup = {1}")
```

Comment:
Parameterized the test to make sure both cleanup strategies work for the existing test cases.

Reply:
Overall, the tests look correct to me. Just a couple minor things but no blockers.